### ISSUE TITLE
o.m.a.eclipse initialization does not find Alf libraries

### ISSUE THREAD

**Issue by JoriDubrovin** [ModelDriven/Alf-Reference-Implementation#14]
I am trying to set up the EMF resources for Libraries/uml/Alf.library.uml and other library files by running
```java
new org.modeldriven.alf.eclipse.units.RootNamespaceImpl().initialize();
```

That seems to run into two issues:
* The constructor ends up calling `setLibraryDirectory` with null `resourceSet`, so pathmaps do not get initialized.  This is an instance of the "calling a non-final method from constructor" problem.
* The `baseURI` is hard coded, so it does not find Libraries in the plugin.

My current attempt at a workaround is
```java
URI baseUri = URI.createURI(Platform.getBundle("org.modeldriven.alf.eclipse").getResource("/").toString());
RootNamespaceImpl rootNs = new RootNamespaceImpl(baseUri);

// Work around org.modeldriven.alf.eclipse.units.RootNamespaceImpl.resourceSet being uninitialized when
// setLibraryDirectory is called from RootNamespaceImpl constructor.
rootNs.setLibraryDirectory(RootNamespaceImpl.DEFAULT_LIBRARY_DIRECTORY);

rootNs.initialize();
```
while patching RootNamespaceImpl with another constructor
```java
public RootNamespaceImpl(URI baseURI) {
    super();
    this.resourceSet = new ResourceSetImpl();
    this.baseURI = baseURI;
    UMLResourcesUtil.init(this.resourceSet);
}
```

Using [v0.6.0g](https://github.com/ModelDriven/Alf-Reference-Implementation/releases/tag/v0.6.0g).

**Comment by seidewitz** [comment]
The first issue is due to setLibraryDirectory being called in the constructor for the superclass org.modeldriven.alf.fuml.RootNamespaceImpl(). This is definitely a bug, which was covered up by the fact that org.modeldriven.alf.eclipse.execution.AlfCompiler has leftover code to explicitly set the default library (much as you did in your workaround.)

The second issue was not a bug from the original intent of the Eclipse implementation, but it is for actually using the implementation as a plugin.

**Comment by JoriDubrovin** [comment]
[v0.6.0h](https://github.com/ModelDriven/Alf-Reference-Implementation/releases/tag/v0.6.0h) works without problems using the new `RootNamespaceImpl(URI)` constructor.

**Comment by seidewitz** [comment]
Resolved in v0.6.0h.
